### ISSUE TITLE
*-origin repo is added even if a flatpakref installation is cancelled

### ISSUE THREAD

**Issue by AsciiWolf** [flatpak/flatpak#4452]
## Linux distribution and version
Fedora 35 Beta

## Flatpak version
flatpak-1.11.3-1.fc35.x86_64

## Description of the problem
When installing, for example, some test build from Flathub using *.flatpakref URL, but pressing "n" when being asked "Proceed with these changes to the user installation?", the installation is correctly cancelled, but *-origin repo that belongs to this flatpakref is kept.

## Steps to reproduce
1. See what remotes are enabled using the `flatpak remotes` command.
2. Try installing some test build from Flathub, for example: `flatpak install --user https://dl.flathub.org/build-repo/60483/net.supertuxkart.SuperTuxKart.flatpakref`

3. Press "n" when being asked "Proceed with these changes to the user installation?".
4. See what remotes are enabled using the `flatpak remotes` command.

**Comment by mwleeds** [comment]
Yeah there's cleanup code at the bottom of `flatpak_transaction_real_run()` to remove the origin remote(s) in case the transaction fails, but we don't hit it if the user aborts, so I would agree that's a bug.
